<h1> Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst:jojo</h3>

In [1]:
import numpy as np
import pandas as pd

<p>Importing dataset to Jupyter notebook</p>
<p>Converting dataset to dataframe</p>
<ul><strong>Requirements</strong>
<li> Name of dataframe</li>
<li>Source</li>
    
</ul>

In [2]:
penguins = pd.read_csv('datasets\\penguins.csv')

In [3]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h3>Descriptive Analytics</h3>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Data Frame</h3>

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2> Selecting a column</h2>
<h3>Syntax</h3><br>
data_frame['column_name']

In [7]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [8]:
penguins['island'] =='Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [9]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [10]:
penguins['species']=="Adelie"

0       True
1       True
2       True
3       True
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Name: species, Length: 344, dtype: bool

<h1>Species Dataframe</h1>

In [11]:
penguins[penguins['species']=="Adelie"]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [12]:
penguins[penguins['species']=="Chinstrap"]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [13]:
penguins[penguins['species']=="Gentoo"]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

<h1>How many does chinstrap weigh above 4000?</h1>

In [14]:
penguins['body_mass_g'] >=4000

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Name: body_mass_g, Length: 344, dtype: bool

In [15]:
(penguins['species']=="Chinstrap") & (penguins['body_mass_g'] >=4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340    False
341    False
342     True
343    False
Length: 344, dtype: bool

In [16]:
penguins[(penguins['species']=="Chinstrap") & (penguins['body_mass_g'] >=4000)]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
284  Chinstrap  Dream            46.0           18.9              195.0   
289  Chinstrap  Dream            52.0           18.1              201.0   
291  Chinstrap  Dream            50.5           19.6              201.0   
295  Chinstrap  Dream            49.2           18.2              195.0   
301  Chinstrap  Dream            52.0           19.0              197.0   
305  Chinstrap  Dream            52.8           20.0              205.0   
307  Chinstrap  Dream            54.2           20.8              201.0   
309  Chinstrap  Dream            51.0           18.8              203.0   
313  Chinstrap  Dream            52.0           20.7              210.0   
315  Chinstrap  Dream            53.5           19.9              205.0   
321  Chinstrap  Dream            50.8           18.5              201.0   
323  Chinstrap  Dream            49.0           19.6              212.0   
329  Chinstrap  Dream            50.7           19.7              203.0   
333  Chinstrap  Dream            49.3           19.9              203.0   
339  Chinstrap  Dream            55.8           19.8              207.0   
342  Chinstrap  Dream            50.8           19.0              210.0   

     body_mass_g     sex  year  
284       4150.0  female  2007  
289       4050.0    male  2007  
291       4050.0    male  2007  
295       4400.0    male  2007  
301       4150.0    male  2007  
305       4550.0    male  2008  
307       4300.0    male  2008  
309       4100.0    male  2008  
313       4800.0    male  2008  
315       4500.0    male  2008  
321       4450.0    male  2009  
323       4300.0    male  2009  
329       4050.0    male  2009  
333       4050.0    male  2009  
339       4000.0    male  2009  
342       4100.0    male  2009

In [17]:
heavy_chinstrap=penguins[(penguins['species']=="Chinstrap") & (penguins['body_mass_g'] >=4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


<h1>How many Adelie female have?</h1>

In [18]:
penguins['sex'] == 'female'

0      False
1       True
2       True
3      False
4       True
       ...  
339    False
340     True
341    False
342    False
343     True
Name: sex, Length: 344, dtype: bool

In [19]:
(penguins['species']=="Adelie") & (penguins['sex'] == 'female')

0      False
1       True
2       True
3      False
4       True
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [20]:
penguins[(penguins['species']=="Adelie") & (penguins['sex'] == 'female')]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
6    Adelie  Torgersen            38.9           17.8              181.0   
12   Adelie  Torgersen            41.1           17.6              182.0   
..      ...        ...             ...            ...                ...   
142  Adelie      Dream            32.1           15.5              188.0   
144  Adelie      Dream            37.3           16.8              192.0   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
150  Adelie      Dream            36.0           17.1              187.0   

     body_mass_g     sex  year  
1         3800.0  female  2007  
2         3250.0  female  2007  
4         3450.0  female  2007  
6         3625.0  female  2007  
12        3200.0  female  2007  
..           ...     ...   ...  
142       3050.0  female  2009  
144       3000.0  female  2009  
147       3475.0  female  2009  
148       3450.0  female  2009  
150       3700.0  female  2009  

[73 rows x 8 columns]

In [21]:
penguins['body_mass_kg'] = np.round(penguins['body_mass_g'] / 1000,1)

In [22]:
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  body_mass_kg  
0         3750.0    male  2007           3.8  
1         3800.0  female  2007           3.8  
2         3250.0  female  2007           3.2  
3            NaN     NaN  2007           NaN  
4         3450.0  female  2007           3.4  
..           ...     ...   ...           ...  
339       4000.0    male  2009           4.0  
340       3400.0  female  2009           3.4  
341       3775.0    male  2009           3.8  
342       4100.0    male  2009           4.1  
343       3775.0  female  2009           3.8  

[344 rows x 9 columns]

In [23]:
adelie_df = penguins[penguins['species'] == 'Adelie']

In [24]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  body_mass_kg  
0         3750.0    male  2007           3.8  
1         3800.0  female  2007           3.8  
2         3250.0  female  2007           3.2  
3            NaN     NaN  2007           NaN  
4         3450.0  female  2007           3.4  
..           ...     ...   ...           ...  
147       3475.0  female  2009           3.5  
148       3450.0  female  2009           3.4  
149       3750.0    male  2009           3.8  
150       3700.0  female  2009           3.7  
151       4000.0    male  2009           4.0  

[152 rows x 9 columns]

In [25]:
gentoo_df = penguins[penguins['species'] == 'Gentoo']

In [26]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  body_mass_kg  
152       4500.0  female  2007           4.5  
153       5700.0    male  2007           5.7  
154       4450.0  female  2007           4.4  
155       5700.0    male  2007           5.7  
156       5400.0    male  2007           5.4  
..           ...     ...   ...           ...  
271          NaN     NaN  2009           NaN  
272       4850.0  female  2009           4.8  
273       5750.0    male  2009           5.8  
274       5200.0  female  2009           5.2  
275       5400.0    male  2009           5.4  

[124 rows x 9 columns]

In [27]:
chinstrap_df = penguins[penguins['species'] == 'Chinstrap']

In [28]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  body_mass_kg  
276       3500.0  female  2007           3.5  
277       3900.0    male  2007           3.9  
278       3650.0    male  2007           3.6  
279       3525.0  female  2007           3.5  
280       3725.0    male  2007           3.7  
..           ...     ...   ...           ...  
339       4000.0    male  2009           4.0  
340       3400.0  female  2009           3.4  
341       3775.0    male  2009           3.8  
342       4100.0    male  2009           4.1  
343       3775.0  female  2009           3.8  

[68 rows x 9 columns]

In [29]:
sex_adelie=penguins[(penguins['species']=="Adelie") & (penguins['sex'] == 'female')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 73 entries, 1 to 150
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            73 non-null     object 
 1   island             73 non-null     object 
 2   bill_length_mm     73 non-null     float64
 3   bill_depth_mm      73 non-null     float64
 4   flipper_length_mm  73 non-null     float64
 5   body_mass_g        73 non-null     float64
 6   sex                73 non-null     object 
 7   year               73 non-null     int64  
 8   body_mass_kg       73 non-null     float64
dtypes: float64(5), int64(1), object(3)
memory usage: 5.7+ KB


In [30]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
body_mass_kg          2
dtype: int64

<h1>Bill_length Filling</h1>

In [31]:
penguins['bill_length_mm'].mean()

np.float64(43.9219298245614)

In [32]:
penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

0      39.10000
1      39.50000
2      40.30000
3      43.92193
4      36.70000
         ...   
339    55.80000
340    43.50000
341    49.60000
342    50.80000
343    50.20000
Name: bill_length_mm, Length: 344, dtype: float64

In [33]:
penguins['bill_length_mm']=penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mean())

In [34]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
body_mass_kg          2
dtype: int64

<h1>Bill_depth filling</h1>

In [35]:
penguins['bill_depth_mm'].mean()

np.float64(17.151169590643274)

In [36]:
penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

0      18.70000
1      17.40000
2      18.00000
3      17.15117
4      19.30000
         ...   
339    19.80000
340    18.10000
341    18.20000
342    19.00000
343    18.70000
Name: bill_depth_mm, Length: 344, dtype: float64

In [37]:
penguins['bill_depth_mm']=penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mean())

In [38]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
body_mass_kg          2
dtype: int64

<h1>Flipper_length filling</h1>

In [39]:
penguins['flipper_length_mm'].mean()

np.float64(200.91520467836258)

In [40]:
penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

0      181.000000
1      186.000000
2      195.000000
3      200.915205
4      193.000000
          ...    
339    207.000000
340    202.000000
341    193.000000
342    210.000000
343    198.000000
Name: flipper_length_mm, Length: 344, dtype: float64

In [41]:
penguins['flipper_length_mm']=penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mean())

In [42]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           2
sex                  11
year                  0
body_mass_kg          2
dtype: int64

<h1>Body_mass Fillling</h1>

In [43]:
penguins['body_mass_g'].mean()

np.float64(4201.754385964912)

In [44]:
penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

0      3750.000000
1      3800.000000
2      3250.000000
3      4201.754386
4      3450.000000
          ...     
339    4000.000000
340    3400.000000
341    3775.000000
342    4100.000000
343    3775.000000
Name: body_mass_g, Length: 344, dtype: float64

In [45]:
penguins['body_mass_g']=penguins['body_mass_g'].fillna(penguins['body_mass_g'].mean())

In [46]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        0
bill_depth_mm         0
flipper_length_mm     0
body_mass_g           0
sex                  11
year                  0
body_mass_kg          2
dtype: int64

<h1>Sex_filling</h1>

In [47]:
penguins['sex'].info()

<class 'pandas.core.series.Series'>
RangeIndex: 344 entries, 0 to 343
Series name: sex
Non-Null Count  Dtype 
--------------  ----- 
333 non-null    object
dtypes: object(1)
memory usage: 2.8+ KB


In [48]:
penguins['sex'].fillna(penguins['sex'].mode())

0        male
1      female
2      female
3         NaN
4      female
        ...  
339      male
340    female
341      male
342      male
343    female
Name: sex, Length: 344, dtype: object

In [49]:
penguins['sex']=penguins['sex'].fillna(penguins['sex'].mode()[0])

In [50]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
body_mass_kg         2
dtype: int64

<h1> How to change datatype </h1>

In [51]:
penguins['species']= penguins['species'].astype('category')
penguins['island']= penguins['island'].astype('category')
penguins['sex']= penguins['sex'].astype('category')



In [52]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
 8   body_mass_kg       342 non-null    float64 
dtypes: category(3), float64(5), int64(1)
memory usage: 17.6 KB


<h1>EXPLANATORY DATA ANALYSIS (EDA)</h1>

<h1>1.How many Chinstrap female have?</h1>

In [53]:
penguins['sex'] == 'female'

0      False
1       True
2       True
3      False
4       True
       ...  
339    False
340     True
341    False
342    False
343     True
Name: sex, Length: 344, dtype: bool

In [54]:
(penguins['species']=="Chinstrap") & (penguins['sex'] == 'female')

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340     True
341    False
342    False
343     True
Length: 344, dtype: bool

In [55]:
penguins[(penguins['species']=="Chinstrap") & (penguins['sex'] == 'female')]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
281  Chinstrap  Dream            45.2           17.8              198.0   
282  Chinstrap  Dream            46.1           18.2              178.0   
284  Chinstrap  Dream            46.0           18.9              195.0   
286  Chinstrap  Dream            46.6           17.8              193.0   
288  Chinstrap  Dream            47.0           17.3              185.0   
290  Chinstrap  Dream            45.9           17.1              190.0   
293  Chinstrap  Dream            58.0           17.8              181.0   
294  Chinstrap  Dream            46.4           18.6              190.0   
296  Chinstrap  Dream            42.4           17.3              181.0   
298  Chinstrap  Dream            43.2           16.6              187.0   
300  Chinstrap  Dream            46.7           17.9              195.0   
302  Chinstrap  Dream            50.5           18.4              200.0   
304  Chinstrap  Dream            46.4           17.8              191.0   
306  Chinstrap  Dream            40.9           16.6              187.0   
308  Chinstrap  Dream            42.5           16.7              187.0   
311  Chinstrap  Dream            47.5           16.8              199.0   
312  Chinstrap  Dream            47.6           18.3              195.0   
314  Chinstrap  Dream            46.9           16.6              192.0   
317  Chinstrap  Dream            46.2           17.5              187.0   
319  Chinstrap  Dream            45.5           17.0              196.0   
320  Chinstrap  Dream            50.9           17.9              196.0   
322  Chinstrap  Dream            50.1           17.9              190.0   
325  Chinstrap  Dream            49.8           17.3              198.0   
326  Chinstrap  Dream            48.1           16.4              199.0   
328  Chinstrap  Dream            45.7           17.3              193.0   
330  Chinstrap  Dream            42.5           17.3              187.0   
332  Chinstrap  Dream            45.2           16.6              191.0   
335  Chinstrap  Dream            45.6           19.4              194.0   
337  Chinstrap  Dream            46.8           16.5              189.0   
338  Chinstrap  Dream            45.7           17.0              195.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  body_mass_kg  
276       3500.0  female  2007           3.5  
279       3525.0  female  2007           3.5  
281       3950.0  female  2007           4.0  
282       3250.0  female  2007           3.2  
284       4150.0  female  2007           4.2  
286       3800.0  female  2007           3.8  
288       3700.0  female  2007           3.7  
290       3575.0  female  2007           3.6  
293       3700.0  female  2007           3.7  
294       3450.0  female  2007           3.4  
296       3600.0  female  2007           3.6  
298       2900.0  female  2007           2.9  
300       3300.0  female  2007           3.3  
302       3400.0  female  2008           3.4  
304       3700.0  female  2008           3.7  
306       3200.0  female  2008           3.2  
308       3350.0  female  2008           3.4  
311       3900.0  female  2008           3.9  
312       3850.0  female  2008           3.8  
314       2700.0  female  2008           2.7  
317       3650.0  female  2008           3.6  
319       3500.0  female  2008           3.5  
320       3675.0  female  2009           3.7  
322       3400.0  female  2009           3.4  
325       3675.0  female  2009           3.7  
326       3325.0  female  2009           3.3  
328       3600.0  female  2009           3.6  
330       3350.0  female  2009           3.4  
332       3250.0  

In [56]:
sex_chinstrap=penguins[(penguins['species']=="Chinstrap") & (penguins['sex'] == 'female')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 34 entries, 276 to 343
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            34 non-null     category
 1   island             34 non-null     category
 2   bill_length_mm     34 non-null     float64 
 3   bill_depth_mm      34 non-null     float64 
 4   flipper_length_mm  34 non-null     float64 
 5   body_mass_g        34 non-null     float64 
 6   sex                34 non-null     category
 7   year               34 non-null     int64   
 8   body_mass_kg       34 non-null     float64 
dtypes: category(3), float64(5), int64(1)
memory usage: 2.3 KB


<h1>1.INSIGHTS</h1>
<H2>The Species Chinstrap have 34 Females</H2>

<h1>2.How many Adelie species are in island Dream?</h1>

In [57]:
penguins['island'] == 'Dream'

0      False
1      False
2      False
3      False
4      False
       ...  
339     True
340     True
341     True
342     True
343     True
Name: island, Length: 344, dtype: bool

In [58]:
(penguins['species']=="Adelie") & (penguins['island'] == 'Dream')

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [59]:
penguins[(penguins['species']=="Adelie") & (penguins['island'] == 'Dream')]

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30   Adelie  Dream            39.5           16.7              178.0   
31   Adelie  Dream            37.2           18.1              178.0   
32   Adelie  Dream            39.5           17.8              188.0   
33   Adelie  Dream            40.9           18.9              184.0   
34   Adelie  Dream            36.4           17.0              195.0   
35   Adelie  Dream            39.2           21.1              196.0   
36   Adelie  Dream            38.8           20.0              190.0   
37   Adelie  Dream            42.2           18.5              180.0   
38   Adelie  Dream            37.6           19.3              181.0   
39   Adelie  Dream            39.8           19.1              184.0   
40   Adelie  Dream            36.5           18.0              182.0   
41   Adelie  Dream            40.8           18.4              195.0   
42   Adelie  Dream            36.0           18.5              186.0   
43   Adelie  Dream            44.1           19.7              196.0   
44   Adelie  Dream            37.0           16.9              185.0   
45   Adelie  Dream            39.6           18.8              190.0   
46   Adelie  Dream            41.1           19.0              182.0   
47   Adelie  Dream            37.5           18.9              179.0   
48   Adelie  Dream            36.0           17.9              190.0   
49   Adelie  Dream            42.3           21.2              191.0   
84   Adelie  Dream            37.3           17.8              191.0   
85   Adelie  Dream            41.3           20.3              194.0   
86   Adelie  Dream            36.3           19.5              190.0   
87   Adelie  Dream            36.9           18.6              189.0   
88   Adelie  Dream            38.3           19.2              189.0   
89   Adelie  Dream            38.9           18.8              190.0   
90   Adelie  Dream            35.7           18.0              202.0   
91   Adelie  Dream            41.1           18.1              205.0   
92   Adelie  Dream            34.0           17.1              185.0   
93   Adelie  Dream            39.6           18.1              186.0   
94   Adelie  Dream            36.2           17.3              187.0   
95   Adelie  Dream            40.8           18.9              208.0   
96   Adelie  Dream            38.1           18.6              190.0   
97   Adelie  Dream            40.3           18.5              196.0   
98   Adelie  Dream            33.1           16.1              178.0   
99   Adelie  Dream            43.2           18.5              192.0   
132  Adelie  Dream            36.8           18.5              193.0   
133  Adelie  Dream            37.5           18.5              199.0   
134  Adelie  Dream            38.1           17.6              187.0   
135  Adelie  Dream            41.1           17.5              190.0   
136  Adelie  Dream            35.6           17.5              191.0   
137  Adelie  Dream            40.2           20.1              200.0   
138  Adelie  Dream            37.0           16.5              185.0   
139  Adelie  Dream            39.7           17.9              193.0   
140  Adelie  Dream            40.2           17.1              193.0   
141  Adelie  Dream            40.6           17.2              187.0   
142  Adelie  Dream            32.1           15.5              188.0   
143  Adelie  Dream            40.7           17.0              190.0   
144  Adelie  Dream            37.3           16.8              192.0   
145  Adelie  Dream            39.0           18.7              185.0   
146  Adelie  Dream            39.2           18.6              190.0   
147  Adelie  Dream            36.6           18.4              184.0   
148  Adelie  Dream            36.0           17.8              195.0   
149  Adelie  Dream            37.8           18.1              193.0   
150  Adelie  Dream            36.0          

In [60]:
sex_chinstrap=penguins[(penguins['species']=="Adelie") & (penguins['island'] == 'Dream')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 56 entries, 30 to 151
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            56 non-null     category
 1   island             56 non-null     category
 2   bill_length_mm     56 non-null     float64 
 3   bill_depth_mm      56 non-null     float64 
 4   flipper_length_mm  56 non-null     float64 
 5   body_mass_g        56 non-null     float64 
 6   sex                56 non-null     category
 7   year               56 non-null     int64   
 8   body_mass_kg       56 non-null     float64 
dtypes: category(3), float64(5), int64(1)
memory usage: 3.6 KB


<H1>2.INSIGHTS</H1>
<H2> There are 56 Adeline that is currently living in island Dream</H2>

<h1>3.How many does Gentoo weigh below 4000?</h1>

In [61]:
penguins['body_mass_g'] <=4000

0       True
1       True
2       True
3      False
4       True
       ...  
339     True
340     True
341     True
342    False
343     True
Name: body_mass_g, Length: 344, dtype: bool

In [62]:
(penguins['species']=="Gentoo") & (penguins['body_mass_g'] <=4000)

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [63]:
penguins[(penguins['species']=="Gentoo") & (penguins['body_mass_g'] <=4000)]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
192  Gentoo  Biscoe            42.7           13.7              208.0   

     body_mass_g     sex  year  body_mass_kg  
192       3950.0  female  2008           4.0

<h1>3. Insights</h1>
<h2>There is only one Gentoo that is weigh below 4000 bodymass and it is located in island Biscoe</h2>

<h1>4.In island Biscoe, how many are Gentoo living there?</h1>

In [64]:
penguins['island'] == 'Biscoe'

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Name: island, Length: 344, dtype: bool

In [65]:
(penguins['species']=="Gentoo") & (penguins['island'] == 'Biscoe')

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [66]:
penguins[(penguins['species']=="Gentoo") & (penguins['island'] == 'Biscoe')]

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe        46.10000       13.20000         211.000000   
153  Gentoo  Biscoe        50.00000       16.30000         230.000000   
154  Gentoo  Biscoe        48.70000       14.10000         210.000000   
155  Gentoo  Biscoe        50.00000       15.20000         218.000000   
156  Gentoo  Biscoe        47.60000       14.50000         215.000000   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe        43.92193       17.15117         200.915205   
272  Gentoo  Biscoe        46.80000       14.30000         215.000000   
273  Gentoo  Biscoe        50.40000       15.70000         222.000000   
274  Gentoo  Biscoe        45.20000       14.80000         212.000000   
275  Gentoo  Biscoe        49.90000       16.10000         213.000000   

     body_mass_g     sex  year  body_mass_kg  
152  4500.000000  female  2007           4.5  
153  5700.000000    male  2007           5.7  
154  4450.000000  female  2007           4.4  
155  5700.000000    male  2007           5.7  
156  5400.000000    male  2007           5.4  
..           ...     ...   ...           ...  
271  4201.754386    male  2009           NaN  
272  4850.000000  female  2009           4.8  
273  5750.000000    male  2009           5.8  
274  5200.000000  female  2009           5.2  
275  5400.000000    male  2009           5.4  

[124 rows x 9 columns]

In [67]:
sex_chinstrap=penguins[(penguins['species']=="Gentoo") & (penguins['island'] == 'Biscoe')].info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 152 to 275
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            124 non-null    category
 1   island             124 non-null    category
 2   bill_length_mm     124 non-null    float64 
 3   bill_depth_mm      124 non-null    float64 
 4   flipper_length_mm  124 non-null    float64 
 5   body_mass_g        124 non-null    float64 
 6   sex                124 non-null    category
 7   year               124 non-null    int64   
 8   body_mass_kg       123 non-null    float64 
dtypes: category(3), float64(5), int64(1)
memory usage: 7.5 KB


<h1>4.Insights</h1>
<h2>There are 124 Gentoo penguins that are living in island Biscoe</h2>

<h1>How many does Adelie weigh above 4000?</h1>

In [68]:
penguins['body_mass_g'] <=4000

0       True
1       True
2       True
3      False
4       True
       ...  
339     True
340     True
341     True
342    False
343     True
Name: body_mass_g, Length: 344, dtype: bool

In [69]:
(penguins['species']=="Adelie") & (penguins['body_mass_g'] >=4500)

0      False
1      False
2      False
3      False
4      False
       ...  
339    False
340    False
341    False
342    False
343    False
Length: 344, dtype: bool

In [70]:
penguins[(penguins['species']=="Adelie") & (penguins['body_mass_g'] >=4500)]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7    Adelie  Torgersen            39.2           19.6              195.0   
17   Adelie  Torgersen            42.5           20.7              197.0   
39   Adelie      Dream            39.8           19.1              184.0   
45   Adelie      Dream            39.6           18.8              190.0   
81   Adelie  Torgersen            42.9           17.6              196.0   
101  Adelie     Biscoe            41.0           20.0              203.0   
109  Adelie     Biscoe            43.2           19.0              197.0   
111  Adelie     Biscoe            45.6           20.3              191.0   

     body_mass_g   sex  year  body_mass_kg  
7         4675.0  male  2007           4.7  
17        4500.0  male  2007           4.5  
39        4650.0  male  2007           4.6  
45        4600.0  male  2007           4.6  
81        4700.0  male  2008           4.7  
101       4725.0  male  2009           4.7  
109       4775.0  male  2009           4.8  
111       4600.0  male  2009           4.6

<h1>INSIGHT #5</h1>
<h1> There are eight Adelie that are weigh above 45000g </h1>

<h1>6.Which species of penguins have the lightiest weight in body mass?</h1>


In [71]:
chinstrap_df['body_mass_kg'].mean()

np.float64(3.730882352941177)

In [72]:
adelie_df['body_mass_kg'].mean()

np.float64(3.7019867549668874)

In [73]:
gentoo_df['body_mass_kg'].mean()

np.float64(5.076422764227642)

<h1>#6 INSIGHTS</h1>
<h1>The lightest species among the three penguins are the Adelie Penguins</h1>

<H1>7.How many chinstrap are recorded in year 2009?</H1>

In [74]:
chinstrap_df['year'].value_counts()

year
2007    26
2009    24
2008    18
Name: count, dtype: int64

<h1>#7 INSIGHTS</h1>
<H1> In year 2009, there are 24 chinstrap that are recorded.</H1>

<h1>Data Aggregation</h1>

<h1>8.How many Adelie have registered to each island in year 2008?</h1>

In [75]:
perisland=penguins.groupby(['species', 'island','year']).size()

C:\Users\TOSHIBA\AppData\Local\Temp\ipykernel_7416\1748534966.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  perisland=penguins.groupby(['species', 'island','year']).size()


In [76]:
perisland

species    island     year
Adelie     Biscoe     2007    10
                      2008    18
                      2009    16
           Dream      2007    20
                      2008    16
                      2009    20
           Torgersen  2007    20
                      2008    16
                      2009    16
Chinstrap  Biscoe     2007     0
                      2008     0
                      2009     0
           Dream      2007    26
                      2008    18
                      2009    24
           Torgersen  2007     0
                      2008     0
                      2009     0
Gentoo     Biscoe     2007    34
                      2008    46
                      2009    44
           Dream      2007     0
                      2008     0
                      2009     0
           Torgersen  2007     0
                      2008     0
                      2009     0
dtype: int64

<h1>INSIGHTS #8</h1>
<H1>There are 18 Adelie that is recorded in island Biscoe, 16 in Dream island and 16 in island Torgersen that were recorded in 2008 </H1>

<h1>9.Average bill_depth_mm of Adelie </h1>

In [77]:
pengbill=penguins.groupby(['species', 'island']) ['bill_depth_mm'].mean()

C:\Users\TOSHIBA\AppData\Local\Temp\ipykernel_7416\1773150043.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  pengbill=penguins.groupby(['species', 'island']) ['bill_depth_mm'].mean()


In [78]:
pengbill

species    island   
Adelie     Biscoe       18.370455
           Dream        18.251786
           Torgersen    18.404830
Chinstrap  Biscoe             NaN
           Dream        18.420588
           Torgersen          NaN
Gentoo     Biscoe       14.999606
           Dream              NaN
           Torgersen          NaN
Name: bill_depth_mm, dtype: float64

<h1>#9 INSIGHTS</h1>
<h1>In island Biscoe the average bill_depth_mm is 18.370455, in island dream is 18.251786 and in island torgersen is 18.404830</h1>

<h1>10.How many species are recorded for each year?</h1>

In [79]:
penguins.groupby('year')['species'].size()

year
2007    110
2008    114
2009    120
Name: species, dtype: int64

<h1>INSIGHTS #10
</h1>
<h1> In year 2007 there are 110 species have been recorded while in year 2008 there are 114 and 120 in year 2009. </h1>

<h1>11.Which species has the lowest average flipper length?</h1>

In [80]:
penguins.groupby('species')['flipper_length_mm'].mean()

C:\Users\TOSHIBA\AppData\Local\Temp\ipykernel_7416\1754789030.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('species')['flipper_length_mm'].mean()


species
Adelie       190.025758
Chinstrap    195.823529
Gentoo       217.055768
Name: flipper_length_mm, dtype: float64

<H1>INSIGHTS #11</H1>
<h1>Adelie have the lowest average flipper length with only 190.025758 average </h1>

<h1>12.What is the average body mass per year?</h1>

In [81]:
penguins.groupby('year')['body_mass_g'].mean()

year
2007    4125.243222
2008    4266.666667
2009    4210.222953
Name: body_mass_g, dtype: float64

<H1>INSIGHTS #12</H1>
<H1>In year 2007 the average body mass is 4125.243222 while in 2008 its 4266.666667 and 4210.222953 in 2009 </H1>

<h1> 13. Does having a long flipper means also having a deeper bill?  </h1>

In [82]:
corrdepth = penguins['bill_length_mm'].corr(penguins['bill_depth_mm'])

In [83]:
corrdepth

np.float64(-0.23505287035553277)

<h1>INSIGHT</h1>
<H1></H1>


<H1>14.Are Females penguins heavier than the male penguuins?  </H1>

In [84]:
penguins.groupby('sex')['body_mass_g'].mean()

C:\Users\TOSHIBA\AppData\Local\Temp\ipykernel_7416\1725385282.py:1: FutureWarning: The default of observed=False is deprecated and will be changed to True in a future version of pandas. Pass observed=False to retain current behavior or observed=True to adopt the future default and silence this warning.
  penguins.groupby('sex')['body_mass_g'].mean()


sex
female    3862.272727
male      4514.684407
Name: body_mass_g, dtype: float64

<h1>INSIGHT

</h1>
<h1>Male penguins are much more heavier than the female penguins</h1>